TerrorConsole/Assets/_Main/Scripts/Objects/HidingPlace.cs

```diff
@@ -0,0 +1,65 @@
+using UnityEngine;
+
+namespace TerrorConsole
+{
+    public class HidingPlace : MonoBehaviour
+    {
+        [SerializeField] private SpriteRenderer _playerSprite;
+        private bool _isPlayerInRange = false;
+        [SerializeField] private GameObject _player;
+        private PlayerController _playerController;
+        private bool _isHiding = false;
+
+
+        private void OnCollisionEnter2D(Collision2D other)
+        {
+            if (other.gameObject.CompareTag("Player"))
+            {
+                _player = other.gameObject;
+                _playerController = _player.GetComponent<PlayerController>();
+                _isPlayerInRange = true;
+            }
+        }
+
+        private void OnCollisionExit2D(Collision2D other)
+        {
+            if (other.gameObject.CompareTag("Player"))
+            {
+                _isPlayerInRange = false;
+                _player = null;
+                _playerController = null;
+            }
+        }
+
+        void Update()
+        {
+            if (_isPlayerInRange && Input.GetKeyDown(KeyCode.E))
```

Comment:
This `Input` call should be added to our `InputManager` as another action, preferably an event so you don't use `Update` here.